app/controllers/application_controller.rb

```diff
@@ -1,2 +1,7 @@
+# frozen_string_literal: true
+
 class ApplicationController < ActionController::Base
+
+  protect_from_forgery
```

Comment:
Here you will need to add something like this https://stackoverflow.com/questions/21478006/how-to-add-username-field-to-devise-gem (check the first answer), in order to allow username in sign up or update.
NOTE: Remember to generate Devise views in order to allow username in sign up.